### PR TITLE
[DRAFT DO NOT MERGE] Add support for simulator data logging

### PR DESCRIPTION
**Pull request by microbit-robert**

### CODE REVIEW

File: src/simulator/Simulator.tsx [hunk 1]
```diff
@@ -89,7 +89,7 @@ const Simulator = ({
                 ref={ref}
                 as="iframe"
                 // This needs changing before we remove the flag.
-                src={`https://stage-python-simulator.microbit.org/simulator.html?color=${encodeURIComponent(
+                src={`https://deploy-preview-36--distracted-dubinsky-fd8a42.netlify.app/simulator.html?color=${encodeURIComponent(
```
Comment (microbit-robert) [inline src/simulator/Simulator.tsx:92]: Change this back before merging!

File: src/simulator/model.ts [hunk 2]
```diff
@@ -20,3 +20,17 @@ export interface EnumSensor {
...
+
+export interface LogData {
+  key: string;
+  value: string | number;
+}
+
+export interface DataLoggingSensor {
+  type: "log";
+  id: string;
+  value: [LogData[]];
```
Comment (microbit-matt-hillsdon) [inline src/simulator/model.ts:32]: Do we mean a singleton array here?

Comment (microbit-robert) [reply src/simulator/model.ts:32]: It's currently not, due to some difficulties I was having, but it probably should be.

File: src/simulator/DataLogging.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    return filteredLog.every((l) => l.value === "");
+  };
+
+  return (
+    <HStack spacing={3}>
+      {minimised ? (
+        <HStack justifyContent="space-between" width="100%">
+          {icon}
+          <Text>{Math.max(sensor.value.length - 1, 0)} rows</Text>
+          <Button size="sm">Download</Button>
```
Comment (microbit-robert) [inline src/simulator/DataLogging.tsx:55]: Currently does nothing.

**Comment by microbit-matt-hillsdon** [comment]
Superseded by #922 which built on this branch. Closing.

**Pull request closed without merging**